chalk-ir/src/lib.rs

```diff
@@ -2385,16 +2675,18 @@ where
     }
 }
 
+/// Utility for converting a list of all the binders into scope
+/// into references to those binders. Simply pair the binders with
+/// the indices, and invoke `to_generic_arg()` on the `(binder,
```

Comment:
Another comment not specific to this PR: I feel like this should have been `(index, binder)` so that `variable_kinds.enumerate()` can be casted directly.

Reply:
Good idea, I'll just include it.